rasa_nlu/extractors/crf_entity_extractor.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import unicode_literals, print_function
+
+import os
+import random
+
+import pathlib
+import warnings
+
+from typing import Optional
+
+from rasa_nlu.components import Component
+from rasa_nlu.extractors import EntityExtractor
+from rasa_nlu.training_data import TrainingData
+import spacy
+from spacy.gold import GoldParse
+import pycrfsuite
+import numpy as np
+from tempfile import NamedTemporaryFile
+import time
+import shutil
+
+
+class CRFEntityExtractor(Component, EntityExtractor):
+    name = "ner_crf"
+
+    context_provides = {
+        "process": ["entities"],
+    }
+
+    output_provides = ["entities"]
+
+    function_dict = {'low': lambda doc: doc[0].lower(), 'title': lambda doc: unicode(doc[0].istitle()),
+                     'word3': lambda doc: doc[0][-3:], 'word2': lambda doc: doc[0][-2:],
+                     'pos': lambda doc: doc[1], 'pos2': lambda doc: doc[1][:2],
+                     'bias': lambda doc: u'bias', 'upper': lambda doc: unicode(doc[0].isupper()),
+                     'digit': lambda doc: unicode(doc[0].isdigit())}
+
+    def __init__(self, ent_tagger=None, nlp=spacy.load('en'), crf_features=None, BILOU_flag=True):
```

Comment:
nlp needs to be moved to the train / test / process methods. (We need to avoid loading anything from spacy anywhere, otherwise we'll end up with multiple instances of it at some point :D)